### PR TITLE
[BOSK-62] fix: Fixed examples for new data types (CPUData, GPUData)

### PR DESCRIPTION
**Pull request by AndreyAgeev**
Added "CPUData" to main deep forest use cases, added example XGBoost for "GPUData" and "CPUData"

### CODE REVIEW

File: bosk/block/auto.py [hunk 1]
```diff
@@ -35,6 +35,9 @@ def _auto_block_impl(cls: Type[BaseBlock]):
 
         """
         fit_argnames = set(cls.fit.__code__.co_varnames[1:cls.fit.__code__.co_argcount])
+        if len(fit_argnames) == 0:  # TODO: Arguments not found for XGBOOST
```
Comment (AndreyAgeev) [inline bosk/block/auto.py:38]: can't find arguments for methods XGBoost

File: examples/deep_forests/gpu/gpu_boosting.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+def make_deep_forest_functional_gpu(executor, **ex_kw):
+    b = FunctionalPipelineBuilder()
+    X, y = b.Input()(), b.TargetInput()()
+    rf_1 = b.XGBClassifier(tree_method="gpu_hist", random_state=42)(X=X, y=y)
+    et_1 = b.ETC(random_state=42)(X=X, y=y)
+    rf_1 = b.MoveTo("GPU")(X=rf_1)
+    et_1 = b.MoveTo("GPU")(X=et_1)
+    concat_1 = b.Concat(['X', 'rf_1', 'et_1'])(X=X, rf_1=rf_1, et_1=et_1)
+    rf_2 = b.XGBClassifier(tree_method="gpu_hist", random_state=42)(X=concat_1, y=y)
```
Comment (andruekonst) [inline examples/deep_forests/gpu/gpu_boosting.py:25]: There is a doubt about XGBClassifier: can it work with JAX gpu data?

Probably, it just applies some trick like `np.asarray(X)`, i.e. downloads the data from GPU to CPU, then returns it back to GPU for ensemble construction, and finally converts data to NumPy CPU array.
So ain't it better not to move data to GPU using `MoveTo("GPU")` and run XGBoost directly on CPU data?

**Review by andruekonst — CHANGES_REQUESTED** [review]

File: examples/deep_forests/gpu/gpu_boosting.py (hunk 2 above)
Comment (AndreyAgeev) [reply examples/deep_forests/gpu/gpu_boosting.py:23]: Removed extra blocks`MoveTo("GPU")`

File: bosk/block/auto.py (hunk 1 above)
Comment (AndreyAgeev) [reply bosk/block/auto.py]: Fixed this moment

**Review by andruekonst — APPROVED** [review]

**Pull request merged**